### PR TITLE
Refactor command execution

### PR DESCRIPTION
**Pull request by corny**
This removes redundant code

### CODE REVIEW

File: zfs/zfs.go [hunk 1]
```diff
@@ -81,64 +91,10 @@ func (z *zfs) Recv(fs string, sendCommand *exec.Cmd) error {
...
 
-}
-func (z *remoteZfs) SendIncremental(fs string, prev, snap string) *exec.Cmd {
-	dialstring := fmt.Sprintf("%s@%s", z.user, z.host)
+func (z *Zfs) SendIncremental(fs string, prev, snap string) *exec.Cmd {
 	snapfqn := fmt.Sprintf("%s@%s", fs, snap)
-	cmd := exec.Command("/usr/bin/ssh", dialstring, "-C", "/sbin/zfs", "send", "-i", prev, snapfqn)
-
-	return cmd
+	return z.exec("-C", "/sbin/zfs", "send", "-i", prev, snapfqn)
```
Comment (piotaixr) [inline zfs/zfs.go]: I think the -C  (that enables ssh compression) should be in zfs/cmd.go at line 13

**Review by piotaixr — CHANGES_REQUESTED** [review]
Except for the comment on zfs.go, seems OK to me

File: zfs/zfs.go (hunk 1 above)
Comment (corny) [reply zfs/zfs.go]: Ok, I added the `-C` flag and improved the error handling.

File: zfs/zfs.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func NewLocal() *Zfs {
 // NewRemote creates a wrapper for remote ZFS commands
 func NewRemote(host string, user string) *Zfs {
 	return &Zfs{
-		exec: RemoteExec(fmt.Sprintf("%s@%s", host, user)),
+		exec: RemoteExecutor(fmt.Sprintf("%s@%s", host, user)),
```
Comment (piotaixr) [inline zfs/zfs.go]: Should be fmt.Sprintf("%s@%s", host, user)

File: zfs/zfs.go (hunk 1 above)
Comment (piotaixr) [reply zfs/zfs.go]: The `-C` Should be removed here as you added it in cmd.go ;-)

File: zfs/zfs.go [hunk 3]
```diff
@@ -2,39 +2,57 @@ package zfs
...
+func NewLocal() *Zfs {
+	return &Zfs{
+		exec: LocalExec,
+	}
+}
+
+// NewRemote creates a wrapper for remote ZFS commands
+func NewRemote(host string, user string) *Zfs {
+	return &Zfs{
+		exec: RemoteExecutor(fmt.Sprintf("%s@%s", host, user)),
```
Comment (piotaixr) [inline zfs/zfs.go]: Should be `fmt.Sprintf("%s@%s", user, host)`

**Review by piotaixr — CHANGES_REQUESTED** [review]

**Comment by corny** [comment]
Thanks for reviewing. I updated my PR.

**Review by piotaixr — APPROVED** [review]

**Comment by piotaixr** [comment]
Thanks for the PR!

**Pull request merged**